internal/controller/metal3.io/baremetalhost_controller.go

```diff
@@ -839,15 +866,16 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 
 	provResult, provID, err := prov.Register(
 		provisioner.ManagementAccessData{
-			BootMode:                   info.host.Status.Provisioning.BootMode,
-			AutomatedCleaningMode:      info.host.Spec.AutomatedCleaningMode,
-			State:                      info.host.Status.Provisioning.State,
-			CurrentImage:               getCurrentImage(info.host),
-			PreprovisioningImage:       preprovImg,
-			PreprovisioningNetworkData: preprovisioningNetworkData,
-			HasCustomDeploy:            hasCustomDeploy(info.host),
-			DisablePowerOff:            info.host.Spec.DisablePowerOff,
-			CPUArchitecture:            getHostArchitecture(info.host),
+			BootMode:                         info.host.Status.Provisioning.BootMode,
+			AutomatedCleaningMode:            info.host.Spec.AutomatedCleaningMode,
+			State:                            info.host.Status.Provisioning.State,
+			CurrentImage:                     getCurrentImage(info.host),
+			PreprovisioningImage:             preprovImg,
+			PreprovisioningNetworkData:       preprovisioningNetworkData,
+			PreprovisioningExtraKernelParams: r.retrievePreprovisioningExtraKernelParamsSpec(info, prov),
```

Comment:
I believe this is actually the only function that we need to pass `PreprovisioningExtraKernelParams` to (see other comments), so once the others are removed we should pass `preprovImg` as a param here instead of calling `getPreprovImage()` a second time.